language/stdlib/modules/block.mvir

```diff
@@ -0,0 +1,48 @@
+module Block {
+
+  resource T {
+    // Height of the current block
+    height: u64,
+  }
+
+  // This can only be invoked by the Association address, and only a single time.
+  // Currently, it is invoked in the genesis transaction
+  public initialize() {
```

Comment:
Since this function is only called in genesis transaction, we might as well just set it as private?

Reply:
Good question. My stance on this has been that although we currently have one big direct write set transaction for genesis, it would be better to have a genesis block consisting of:

- An minimal direct write set transaction: (e.g., just the publish core modules that we need to execute any transaction + create a single account)
- A series of ordinary transactions that instantiate the rest of the system state (e.g., publish the market cap resource, publish the block metadata resource, ...)

The advantage of doing it this way is better transparency + making it easier for our verification tools to reason about what the initial state looks like without hardcoding anything.

A long-winded way of saying: I think it might make our life easier in the future if we keep this `public`, but ensure that it can only be published under a single address.